diff --git a/src/lauxlib.c b/src/lauxlib.c
index 923105e..571e959 100644
--- a/src/lauxlib.c
+++ b/src/lauxlib.c
@@ -1115,6 +1115,15 @@ LUALIB_API lua_State *luaL_newstate (void) {
 }
 
 
+LUALIB_API lua_State *luaL_newstate_seed (unsigned int seed) {
+  lua_State *L = lua_newstate_seed(l_alloc, NULL, seed);
+  if (l_likely(L)) {
+    lua_atpanic(L, &panic);
+    lua_setwarnf(L, warnfoff, L);
+  }
+  return L;
+}
+
 LUALIB_API void luaL_checkversion_ (lua_State *L, lua_Number ver, size_t sz) {
   lua_Number v = lua_version(L);
   if (sz != LUAL_NUMSIZES)  /* check numeric types */
